Lib/glyphsLib/builder/sources.py

```diff
@@ -53,6 +53,16 @@ def _to_designspace_source(self, master, is_regular):
         source.filename = build_ufo_path(
             '', source.familyName, source.styleName)
 
+        # Make sure UFO filenames are unique, lest we overwrite masters that
+        # happen to have the same weight name.
+        if any(s != source and s.filename == source.filename
+               for s in self._sources.values()):
+            raise ValueError(
+                "The source Glyphs file has duplicate master "
+                "names or styles. A master name must be unique. Below "
+                "Glyphs 2.5, you can add custom names in the custom name "
+                "field.")
```

Comment:
the error message should at least tell which master names are duplicate.
Also I don't understand when it says "below Glyphs 2.5, you can add custom names in the custom name field."?
Do you mean the "Master Name" custom parameter? If so, this needs to be made clear.
And what about equal or above 2.5?

Again, I think you want to check `s is not source` to make it explicit that your are checking that objects are non identical, not equivalent. Even if the SourceDescriptor class doesn't implement `__eq__` (yet?) and thus falls back on `object.__eq__` which compares objects' id, explicit is better than implicit.